### PR TITLE
Notes for March 23

### PR DESCRIPTION
**Pull request by va6996**

### CODE REVIEW

File: mar23/va2083.md [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+Provide control abstraction so that tenants can manage their own logical network. There is no fixed standard for the control plane. However, at the low level, there is always a datapath involved. There is a packet processing pipeline, a set of flow tables programmed by the control plane. This forms the basis of control abstraction. The end goal is to create a logical datapath that contains a set of logical flow tables so that tenant control plane can control the logical datapath with logical flow entries that actually implement the forwarding behaviour within the virtual network.
+
+### Datapath
+When a packet traverses the network, it traverses a sequence of table pipeline/datapaths. The previous datapath makes the forwarding decision for which datapath to traverse next. This datapaths are usually routers and switches. Hence, to provide arbitrary logical networks to tenants, we need to provide this virtual datapath and the ability to interconnect multiple datapaths. This allows the tenant to form arbitrary network topologies as per requirement.
```
Comment (kc2853) [inline mar23/va2083.md]: "This datapaths are"

File: mar23/va2083.md [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+### Implementation
+Interconnects VMs using IP tunneling using virtual switches on the VMs.
+These virtual switches are programmed by the controller cluster.
+Controller cluster also runs the tenant specific control plane which program the logical datapaths provided by network hypervisor.
+Only IP connectivity is requested from the physical network.
+The source hypervisor tunnels the packet directly to the destination hypervisor. There is no hypervisor processing the packet in the path.
+
+### Virtual switches
+Virtual switch on the hypervisor responsible to implement complete logical network, datapatha and flow processing.
```
Comment (kc2853) [inline mar23/va2083.md]: "datapatha" typo

File: mar23/va2083.md [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+
+These optimization works because datacenter networks are typically uncongested.
+
+### Packet loss
+eRPC keeps things simple by treating re-ordered packets as losses and dropping them (as do current RDMA NICs). When a client suspects a lost packet, it rolls back the request’s wire protocol state using a ‘go-back-N’ mechanism. It reclaims credits and retransmits from the rollback point.
+
+# Limitations
+## Network Virtualization
+* Open flow is very expensive and is a scaling bottleneck.
+* The datacenters need to maintain additional hardware to support network virtualization and need to perform coordinations with the controller cluster. Upon failure of the coordination cluster, the whole datacenter would be impacted.
```
Comment (kc2853) [inline mar23/va2083.md]: I thought the sharding mechanism employed for availability (Section 5.1) covers this, or was that something else?

**Review by kc2853 — COMMENTED** [review]
Appreciate the notes

File: mar23/va2083.md (hunk 3 above)
Comment (va6996) [reply mar23/va2083.md]: Yes you're right, it was for availability. Fixing it in the revision.

**Pull request merged**